src/test/java/org/minions/devfund/katerinaanzoleaga/battleship/ShipTest.java

```diff
@@ -0,0 +1,280 @@
+        assertEquals(false, destroyer.shootAt(1, 0));
+        assertEquals(true, destroyer.shootAt(0, 2));
+        assertEquals(false, destroyer.isSunk());
+        assertEquals(false, destroyer.shootAt(0, destroyer.getLength() + 1));
+        assertEquals(false, destroyer.isSunk());
+        assertEquals(true, destroyer.shootAt(0, destroyer.getLength()));
+        assertEquals(false, destroyer.isSunk());
+        assertEquals(true, destroyer.shootAt(0, destroyer.getLength() - 1));
+        assertEquals(true, destroyer.isSunk());
+
+        theOcean.print();
+    }
+
+    /**
+     * Test several shots until sunk a submarine.
+     */
+    @Test
+    public void testSunkSubmarine() {
+        final int row = 0;
+        final int column = 0;
+        final int columnSubmarineZero = 0;
+        final int columnSubmarineOne = 1;
+        final int columnSubmarineTwo = 2;
+        final Ship ship = new Submarine();
+        ship.placeShipAt(row, column, true, theOcean);
+        ship.shootAt(row, columnSubmarineZero);
+        ship.shootAt(row, columnSubmarineOne);
+        ship.shootAt(row, columnSubmarineTwo);
+        ship.shootAt(row, columnSubmarineZero);
+        assertTrue(ship.isSunk());
+    }
+    /**
+     * Test several shots until sunk a vertical destroyer.
+     */
+    @Test
+    public void testSunkDestroyerH() {
+        final Ship destroyer = new Destroyer();
+        destroyer.placeShipAt(0, 0, true, theOcean);
+        for (int i = 0; i < destroyer.getLength(); ++i) {
+            destroyer.shootAt(0, i);
+            if (i != destroyer.getLength() - 1) {
+                assertFalse(destroyer.isSunk());
+            }
+        }
+        assertTrue(destroyer.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk an horizontal destroyer.
+     */
+    @Test
+    public void testSunkDestroyerV() {
+        final Ship destroyer = new Destroyer();
+        destroyer.placeShipAt(0, 0, false, theOcean);
+        for (int i = 0; i < destroyer.getLength(); ++i) {
+            destroyer.shootAt(i, 0);
+            if (i != destroyer.getLength() - 1) {
+                assertFalse(destroyer.isSunk());
+            }
+        }
+        assertTrue(destroyer.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk an horizontal lightCruiser.
+     */
+    @Test
+    public void testSunkLightCruiserH() {
+        final Ship lightCruiser = new Destroyer();
+        lightCruiser.placeShipAt(0, 0, true, theOcean);
+        for (int i = 0; i < lightCruiser.getLength(); ++i) {
+            lightCruiser.shootAt(0, i);
+            if (i != lightCruiser.getLength() - 1) {
+                assertFalse(lightCruiser.isSunk());
+            }
+        }
+        assertTrue(lightCruiser.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk a vertical lightcruiser.
+     */
+    @Test
+    public void testSunkLightCruiserV() {
+        final Ship lightCruiser = new LightCruiser();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "lightCruiser" which hides the field declared at line 21. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)